src/main/java/it/gov/pagopa/payhub/activities/activity/classifications/IufClassificationActivityImpl.java

```diff
@@ -0,0 +1,89 @@
+package it.gov.pagopa.payhub.activities.activity.classifications;
+
+import it.gov.pagopa.payhub.activities.dao.ClassifyDao;
+import it.gov.pagopa.payhub.activities.dao.PaymentsReportingDao;
+import it.gov.pagopa.payhub.activities.dao.TreasuryDao;
+import it.gov.pagopa.payhub.activities.dto.classifications.ClassifyDTO;
+import it.gov.pagopa.payhub.activities.dto.classifications.IufClassificationActivityResult;
+import it.gov.pagopa.payhub.activities.dto.paymentsreporting.PaymentsReportingDTO;
+import it.gov.pagopa.payhub.activities.dto.treasury.TreasuryDTO;
+import it.gov.pagopa.payhub.activities.exception.PaymentsClassificationSaveException;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+
+/**
+ * Interface for defining an activity to process payment reporting classifications based on IUF.
+ */
+@Slf4j
+@Lazy
+@Component
+public class IufClassificationActivityImpl implements IufClassificationActivity {
+    private final PaymentsReportingDao paymentsReportingDao;
+    private final TreasuryDao treasuryDao;
+    private final ClassifyDao classifyDao;
+
+    private static final String TES_NO_MATCH = "TES_NO_MATCH";
+
+    public IufClassificationActivityImpl(PaymentsReportingDao paymentsReportingDao, TreasuryDao treasuryDao, ClassifyDao classifyDao) {
+        this.paymentsReportingDao = paymentsReportingDao;
+        this.treasuryDao = treasuryDao;
+        this.classifyDao = classifyDao;
+    }
+
+    /**
+     * Processes IUF classification based on the provided parameters.
+     *
+     * @param organizationId the unique identifier of the organization
+     * @param iuf            the unique identifier of the payment reporting flow (IUF)
+     * @return IufClassificationActivityResult containing PaymentsReportingDTO list and success flag
+     */
+    @Override
+    public IufClassificationActivityResult classify(Long organizationId, String iuf) {
+        if (organizationId==null || organizationId.equals(0L))
+            throw new PaymentsClassificationSaveException("organization id may be not null or zero");
+        if (iuf==null || iuf.isEmpty())
+            throw new PaymentsClassificationSaveException("iuf may be not null or blank");
+
+        IufClassificationActivityResult iufClassificationActivityResult = new IufClassificationActivityResult();
+        List<PaymentsReportingDTO> paymentsReportingDTOS = paymentsReportingDao.findByOrganizationIdFlowIdentifierCode(organizationId, iuf);
+
+        for (TreasuryDTO treasuryDTO : treasuryDao.searchByIuf(iuf)) {
```

Comment:
why are searching on treasury?
a you can see in https://pagopa.atlassian.net/wiki/spaces/SPAC/pages/1339031693/Classificazione+incassi#3.5.2.-Classificazione-IUF you should just search on paymentsReporting.
Who will call this classification will perform it because it has stored a treasury, we have not value on fetch it